--- a/config/config.go
+++ b/config/config.go
@@ -18,7 +18,7 @@
 	"regexp"
 	"time"
 
-	"github.com/golang/protobuf/proto"
+	"code.google.com/p/goprotobuf/proto"
 
 	clientmodel "github.com/prometheus/client_golang/model"
 
--- a/config/load.go
+++ b/config/load.go
@@ -16,7 +16,7 @@
 import (
 	"io/ioutil"
 
-	"github.com/golang/protobuf/proto"
+	"code.google.com/p/goprotobuf/proto"
 
 	pb "github.com/prometheus/prometheus/config/generated"
 )
--- a/retrieval/targetmanager_test.go
+++ b/retrieval/targetmanager_test.go
@@ -17,7 +17,7 @@
 	"testing"
 	"time"
 
-	"github.com/golang/protobuf/proto"
+	"code.google.com/p/goprotobuf/proto"
 
 	clientmodel "github.com/prometheus/client_golang/model"
 
