Index: prometheus/web/blob/blob.go
===================================================================
--- prometheus.orig/web/blob/blob.go
+++ prometheus/web/blob/blob.go
@@ -25,7 +25,9 @@ var mimeMap = map[string]string{
 
 // GetFile retrieves the content of an embedded file.
 func GetFile(bucket string, name string) ([]byte, error) {
-	blob, ok := files[bucket][name]
+	//blob, ok := files[bucket][name]
+        var blob []byte = nil
+        ok := false
 	if !ok {
 		return nil, fmt.Errorf("could not find %s/%s (missing or updated files.go?)", bucket, name)
 	}
Index: prometheus/web/web.go
===================================================================
--- prometheus.orig/web/web.go
+++ prometheus/web/web.go
@@ -42,7 +42,9 @@ var (
 	listenAddress  = flag.String("web.listen-address", ":9090", "Address to listen on for the web interface, API, and telemetry.")
 	hostname       = flag.String("web.hostname", "", "Hostname on which the server is available.")
 	metricsPath    = flag.String("web.telemetry-path", "/metrics", "Path under which to expose metrics.")
-	useLocalAssets = flag.Bool("web.use-local-assets", false, "Read assets/templates from file instead of binary.")
+	// useLocalAssets = flag.Bool("web.use-local-assets", false, "Read assets/templates from file instead of binary.")
+        false_val      = false
+	useLocalAssets = &false_val
 	userAssetsPath = flag.String("web.user-assets", "", "Path to static asset directory, available at /user.")
 	enableQuit     = flag.Bool("web.enable-remote-shutdown", false, "Enable remote service shutdown.")
 )
Index: prometheus/console_libraries/prom.lib
===================================================================
--- prometheus.orig/console_libraries/prom.lib
+++ prometheus/console_libraries/prom.lib
@@ -1,14 +1,13 @@
 {{/* vim: set ft=html: */}}
 {{/* Load Prometheus console library JS/CSS. Should go in <head> */}}
 {{ define "prom_console_head" }}
-<link type="text/css" rel="stylesheet" href="{{ pathPrefix }}/static/vendor/rickshaw/rickshaw.min.css">
-<link type="text/css" rel="stylesheet" href="{{ pathPrefix }}/static/vendor/bootstrap-3.3.1/css/bootstrap.min.css">
+<link type="text/css" rel="stylesheet" href="{{ pathPrefix }}/static/rickshaw/rickshaw.min.css">
+<link type="text/css" rel="stylesheet" href="{{ pathPrefix }}/static/bootstrap/css/bootstrap.min.css">
 <link type="text/css" rel="stylesheet" href="{{ pathPrefix }}/static/css/prom_console.css">
-<script src="{{ pathPrefix }}/static/vendor/rickshaw/vendor/d3.v3.js"></script>
-<script src="{{ pathPrefix }}/static/vendor/rickshaw/vendor/d3.layout.min.js"></script>
-<script src="{{ pathPrefix }}/static/vendor/rickshaw/rickshaw.min.js"></script>
-<script src="{{ pathPrefix }}/static/vendor/js/jquery.min.js"></script>
-<script src="{{ pathPrefix }}/static/vendor/bootstrap-3.3.1/js/bootstrap.min.js"></script>
+<script src="{{ pathPrefix }}/static/d3/d3.js"></script>
+<script src="{{ pathPrefix }}/static/rickshaw/rickshaw.min.js"></script>
+<script src="{{ pathPrefix }}/static/jquery/jquery.min.js"></script>
+<script src="{{ pathPrefix }}/static/bootstrap/js/bootstrap.min.js"></script>
 
 <script>var PATH_PREFIX = "{{ pathPrefix }}";</script>
 <script src="{{ pathPrefix }}/static/js/prom_console.js"></script>
Index: prometheus/web/templates/graph.html
===================================================================
--- prometheus.orig/web/templates/graph.html
+++ prometheus/web/templates/graph.html
@@ -1,18 +1,17 @@
 {{define "head"}}
-    <script src="{{ pathPrefix }}/static/vendor/bootstrap-3.3.1/js/bootstrap.min.js"></script>
+    <script src="{{ pathPrefix }}/static/bootstrap/js/bootstrap.min.js"></script>
 
     <link type="text/css" rel="stylesheet" href="{{ pathPrefix }}/static/css/graph.css">
 
-    <link type="text/css" rel="stylesheet" href="{{ pathPrefix }}/static/vendor/rickshaw/rickshaw.min.css">
+    <link type="text/css" rel="stylesheet" href="{{ pathPrefix }}/static/rickshaw/rickshaw.min.css">
     <link type="text/css" rel="stylesheet" href="{{ pathPrefix }}/static/vendor/bootstrap-datetimepicker/bootstrap-datetimepicker.min.css">
 
-    <script src="{{ pathPrefix }}/static/vendor/rickshaw/vendor/d3.v3.js"></script>
-    <script src="{{ pathPrefix }}/static/vendor/rickshaw/vendor/d3.layout.min.js"></script>
-    <script src="{{ pathPrefix }}/static/vendor/rickshaw/rickshaw.min.js"></script>
+    <script src="{{ pathPrefix }}/static/rickshaw/vendor/d3.js"></script>
+    <script src="{{ pathPrefix }}/static/rickshaw/rickshaw.min.js"></script>
     <script src="{{ pathPrefix }}/static/vendor/bootstrap-datetimepicker/bootstrap-datetimepicker.js"></script>
     <script src="{{ pathPrefix }}/static/vendor/bootstrap3-typeahead/bootstrap3-typeahead.min.js"></script>
 
-    <script src="{{ pathPrefix }}/static/vendor/js/handlebars.js"></script>
+    <script src="{{ pathPrefix }}/static/handlebars/handlebars.js"></script>
     <script src="{{ pathPrefix }}/static/vendor/js/jquery.selection.js"></script>
     <script src="{{ pathPrefix }}/static/vendor/js/jquery.hotkeys.js"></script>
 
Index: prometheus/web/templates/_base.html
===================================================================
--- prometheus.orig/web/templates/_base.html
+++ prometheus/web/templates/_base.html
@@ -3,9 +3,9 @@
   <head>
     <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
     <title>Prometheus Time Series Collection and Processing Server</title>
-    <script src="{{ pathPrefix }}/static/vendor/js/jquery.min.js"></script>
+    <script src="{{ pathPrefix }}/static/jquery/jquery.min.js"></script>
 
-    <link type="text/css" rel="stylesheet" href="{{ pathPrefix }}/static/vendor/bootstrap-3.3.1/css/bootstrap.min.css">
+    <link type="text/css" rel="stylesheet" href="{{ pathPrefix }}/static/bootstrap/css/bootstrap.min.css">
     <link type="text/css" rel="stylesheet" href="{{ pathPrefix }}/static/css/prometheus.css">
 
     <script>var PATH_PREFIX = "{{ pathPrefix }}";</script>
