--- a/retrieval/target_test.go
+++ b/retrieval/target_test.go
@@ -139,6 +139,7 @@
 }
 
 func TestTargetScrapeWithFullChannel(t *testing.T) {
+	t.Skip("Skipping time-dependant test.")
 	server := httptest.NewServer(
 		http.HandlerFunc(
 			func(w http.ResponseWriter, r *http.Request) {
--- a/storage/local/storage_test.go
+++ b/storage/local/storage_test.go
@@ -1131,6 +1131,7 @@
 }
 
 func testEvictAndLoadChunkDescs(t *testing.T, encoding chunkEncoding) {
+	t.Skip("Skipping time-dependant test.")
 	samples := make(clientmodel.Samples, 10000)
 	for i := range samples {
 		samples[i] = &clientmodel.Sample{
